.github/workflows/pull-request.yaml

```diff
@@ -66,3 +66,26 @@ jobs:
 
       - name: Cleanup
         run: kind delete cluster
+
+  e2e-envoy-grpc:
+    name: Envoy gRPC End-to-End Test
+    runs-on: ubuntu-18.04
+    steps:
+      - name: Check out code
+        uses: actions/checkout@v2
+
+      - name: Build docker image
+        run: make image tag-latest
+
+      - name: Build testsrv docker image
+        run: make testsrv-image
+        working-directory: examples/grpc
+
+      - name: Run test
+        run: make test-setup test
+        working-directory: examples/grpc
+
+      - name: Run test log dump and cleanup
+        run: make test-teardown
+        if: ${{ always() }}
+        working-directory: examples/grpc
```

Comment:
It would be great if we could find a way to avoid all the duplication. We could create a make target but probably won't look that great in the UI.

Reply:
Totally agreed! I think github actions have a feature for that, the one or two steps that are specific to each variant (PR check, post-merge) can probably be `if:`'ed accordingly.